src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -562,6 +562,15 @@ void network::allocate_primitives() {
                         auto& eltw_mem = eltw_inst->output_memory();
                         auto new_mem = eltw_mem.get_engine()->reinterpret_buffer(eltw_mem, node->get_output_layout());
                         prim_inst->set_output_memory(new_mem);
+                        // When the user of this primitive can be optimized out, the output memory of that should be replaced to first sum buffer.
+                        for (auto& user : prim_inst->get_users()) {
+                            if (user->can_be_optimized()) {
+                                auto& user_inst = _primitives.at(user->id());
+                                auto& user_node = user_inst->get_node();
+                                auto new_user_mem = new_mem->get_engine()->reinterpret_buffer(*new_mem, user_node.get_output_layout());
+                                user_inst->set_output_memory(new_user_mem);
```

Comment:
1) What if it is reorder-reorder-reorder? I think we should be iterative.
2) What if it is not reorder, but implicit concat? In this case, we cannot set output memory like this.

Reply:
* Could you explain why it is causing segfault?